### ISSUE TITLE
Add support for webp source images

### ISSUE THREAD

**Issue by viktor02** [xemle/home-gallery#88]
Hi. For some reason after `./gallery run init -c config.yml` and `./gallery run server -c config.yml`, I have files found but not added to the database(and most likely no requests to the api server, but I'm not sure).

I also tried without native tools(before this log), but the result is the same.

*Logs*: https://pastebin.com/Enu6LWLe
*Files in my library*: .webp  and .mp4
*Config (from cli)*:
```
baseDir: /opt/home-gallery/
configDir: /opt/home-gallery//config
cacheDir: /opt/home-gallery//cache/
sources:
  - index: /opt/home-gallery//config/PhotoArchive.idx
    offline: false
    excludeIfPresent: .galleryignore
    dir: /srv/big/photos/PhotoArchive
extractor:
  apiServer:
    url: http://192.168.10.2:3001
    concurrent: 5
  useNative:
    - convert
    - ffprobe
    - ffmpeg
storage:
  dir: /opt/home-gallery//cache//storage
database:
  file: /opt/home-gallery//config/database.db
  maxMemory: 4096
events:
  file: /opt/home-gallery//config/events.db
server:
  host: 0.0.0.0
  port: 3000
  openBrowser: true
  basePath: /
  watchSources: true
logger:
  - type: console
    level: trace
  - type: file
    level: debug
    file: /opt/home-gallery//config/gallery.log
configPrefix: ""
```

**Comment by xemle** [comment]
Hi @viktor02 

thank you for using HomeGallery and for reaching out.

Regarding to your logs: I can only guess which steps you tried. 

Guess: You are trying to run the import after some previous actions:

The file index has entries and all your files are indexed.

`4. index File index was saved to /opt/home-gallery//config/PhotoArchive.idx and 29508 entries have new checkums/ids 0.96s`

The files are somehow processed via the extractor

`20. cli.extract Extract all meta data and calculated all preview files from 29507 entries 3.3s`

but the database has no entries

`26. database.build Created 0 database entries 0.97s`

Than you start the server

`33. cli.spawn Run cli with serve`

which starts the update of the database as well (since 1.10.0) by default. The update processes only the diffs from the index. The file index is filled up so there is no update to be expected.

Up to now I do not have an idea what could fail. You can try to

- Remove the tailing slashes in the config. From `baseDir` and `cacheDir`
- Run a full import with `./gallery -l trace run import` to see what happens in the extractor phase
- Remove the file index `/opt/home-gallery//config/PhotoArchive.idx` and just start the server. Via `./gallery -l trace run server`. The import should take place. See also the docs for [I want to start all over, how do I reset the gallery?](https://docs.home-gallery.org/faq/index.html#i-want-to-start-all-over-how-do-i-reset-the-gallery)
- Check that the storage directory is filled up. Check files in `/opt/home-gallery/cache/storage`

If the database does not get filled, please send again that log. Maybe the [internals](https://docs.home-gallery.org/internals/index.html) of the gallery might help to know what phase does what.


Since yesterday version 1.11.1 is out. The changes should not solve your problem (see the [changelog](https://github.com/xemle/home-gallery/blob/master/CHANGELOG.md)). But it is always good to have a new version.

**Comment by viktor02** [comment]
So, I deleted everything except binary gallery file and config.yml, removed extra slash from config and run `./gallery -l trace run server -c config.yml`
And the same.

-------

Since the log is too large due to the large number of files, I will try to do the same thing, but only for ~50 files in a folder in the working directory.
1. `mkdir /opt/home-simple && cd /opt/home-simple`
2. `mkdir photos` - and copied test webp files into it
3. `./gallery run init --source photos/ --config config.yml`
4. changed only baseDir to point into `/opt/home-simple`  in config.yml
5. Run `./gallery -l trace run server -c config.yml `
6. No files. Logs is attached
[log.txt](https://github.com/xemle/home-gallery/files/12264928/log.txt)
[gallery.log](https://github.com/xemle/home-gallery/files/12264943/gallery.log)

**Comment by viktor02** [comment]
> Since yesterday version 1.11.1 is out. The changes should not solve your problem (see the [changelog](https://github.com/xemle/home-gallery/blob/master/CHANGELOG.md)). But it is always good to have a new version.

I couldn't find the latest version [here](https://dl.home-gallery.org/dist/latest/), and I don't want to build it myself from sources. Maybe later I'll try to use the image in the docker

**Comment by xemle** [comment]
> > Since yesterday version 1.11.1 is out. ...
> 
> I couldn't find the latest version [here](https://dl.home-gallery.org/dist/latest/), and I don't want to build it myself from sources. Maybe later I'll try to use the image in the docker

Thank you for the hint. The release building is fixed and 1.12.0 is out.

**Comment by xemle** [comment]
Thank you very much for providing the logs. Now I could identify the error.

The webp file extension was not identified as image type. This is now fixed in c445f28.

Please test the new master binary from [master-latest](https://dl.home-gallery.org/dist/unstable/latest/). Either run a full import with `./gallery run import` or reset it again by deleting your file index and start just the server via `./gallery run server`. I recommend the server version.

With your help, the gallery got a bit better. Thank you.

**Comment by viktor02** [comment]
> Thank you very much for providing the logs. Now I could identify the error.
> 
> The webp file extension was not identified as image type. This is now fixed in [c445f28](https://github.com/xemle/home-gallery/commit/c445f28478ffad850ea1e04abead2573d5f8c497).
> 
> Please test the new master binary from [master-latest](https://dl.home-gallery.org/dist/unstable/latest/). Either run a full import with `./gallery run import` or reset it again by deleting your file index and start just the server via `./gallery run server`. I recommend the server version.
> 
> With your help, the gallery got a bit better. Thank you.

Thank you! Now it works!

**Comment by xemle** [comment]
> Thank you! Now it works!

Awesome. Enjoy the gallery and your memories.

**Comment by xemle** [comment]
The fix is rolled out with bugfix release v1.12.1
